relayer/chains/substrate/provider.go

```diff
@@ -172,17 +177,23 @@ func (sp *SubstrateProvider) Timeout() string {
 	return sp.Config.Timeout
 }
 
-func (sp *SubstrateProvider) TrustingPeriod(ctx context.Context) (time.Duration, error) {
-	// TODO: implement a proper trusting period
-	return time.Duration(math.MaxInt), nil
+func (sp *SubstrateProvider) TrustingPeriod(_ context.Context) (time.Duration, error) {
+	switch sp.Config.RelayChain {
+	case int32(beefyclienttypes.RelayChain_POLKADOT):
+		return beefyclienttypes.RelayChain_POLKADOT.TrustingPeriod(), nil
+	case int32(beefyclienttypes.RelayChain_KUSAMA):
+		return beefyclienttypes.RelayChain_KUSAMA.TrustingPeriod(), nil
+	default:
+		return math.MaxInt, nil
```

Comment:
Returning `MaxInt` could be problematic since the trusting period needs to be less than the unbonding period (with the standard being that the trusting period is set to 2/3 of the unbonding period). This would open up an attack vector in the same class as long range attacks where a malicious validator could unbond their stake and then update the light client with a consensus state from before they unbonded.

I would suggest we return an empty value here along with an error indicating that something has gone wrong.

Reply:
Ahh interesting. Well that makes sense, I think this is valid then